Cognoma.py

```diff
@@ -0,0 +1,656 @@
+
+        #######################################################################
+        # For calculating the t-test between the MSE of real and generated data
+
+        # Find all TP53 wildtype examples and their gene expressions
+        SNPpresent = np.array(rightDomain[0, :])
+        SNPpresentExp = np.array(leftDomain[0, :])
+
+        for x in range(rightDomain.shape[0]):
+            if rightDomain[x, self.mutationID] == 0:
+                SNPpresent = np.vstack((SNPpresent, rightDomain[x, :]))
+                SNPpresentExp = np.vstack(
+                    (SNPpresentExp, leftDomain[x, :]))
+
+        # Find all TP53 mutated examples and their gene expressions
+        SNPabsent = np.array(rightDomain[0, :])
+        SNPabsentExp = np.array(leftDomain[0, :])
+
+        for x in range(rightDomain.shape[0]):
+            if rightDomain[x, self.mutationID] == 1:
+                SNPabsent = np.vstack((SNPabsent, rightDomain[x, :]))
+                SNPabsentExp = np.vstack((SNPabsentExp, leftDomain[x, :]))
+
+        # Pick a random grouping of these examples
+        randIndexes = np.random.randint(
+            0, min(SNPpresent.shape[0], SNPabsent.shape[0]) - 1,
+            (n,))
+        SNPpresent = SNPpresent[1:]
+        SNPpresentExp = SNPpresentExp[1:]
+        SNPpresent = SNPpresent[randIndexes]
+        SNPpresentExp = SNPpresentExp[randIndexes]
+
+        SNPabsent = SNPabsent[1:]
+        SNPabsentExp = SNPabsentExp[1:]
+        SNPabsent = SNPabsent[randIndexes]
+        SNPabsentExp = SNPabsentExp[randIndexes]
+
+        # Predict based on the mutations
+        (SNPpresentToExp, _) = modelHandle.rightToLeftModel.predict(
+                                                              SNPpresent)
+
+        # Find the Mean Squared Error
+        SNPpresentToExpMSE = np.square(np.subtract(
+            SNPpresentExp, SNPpresentToExp)).mean(axis=0)
+
+        # Create arrays for induced samples
+        InducedMutation = copy.copy(SNPpresent)
+        InducedMutationExp = copy.copy(SNPpresentExp)
+
+        # Induce a TP53 mutation
+        InducedMutation[:, self.mutationID] = 1
+
+        # Predict based on this mutation
+        (InducedMutationToExp, _) = modelHandle.rightToLeftModel.predict(
+                                                             InducedMutation)
+
+        # Find the Mean Squared Error
+        InducedMutationToExpMSE = np.square(np.subtract(
+            InducedMutationExp, InducedMutationToExp)).mean(axis=0)
+
+        # Run a t test on the MSE's
+        ttest_results = scipy.stats.ttest_ind(
+            SNPpresentToExpMSE, InducedMutationToExpMSE)
+        t_stat = ttest_results.statistic
+        p_val = ttest_results.pvalue
+
+        # Print a table with the t test results
+        table_data = dict(values=[[t_stat], [p_val]])
+        table_labels = dict(values=['Stat', 'PVal'])
+        table = [go.Table(cells=table_data, header=table_labels)]
+
+        py.offline.plot(table,
```

Comment:
instead of plotting the table, lets write it to a text file instead. This can be done easily with pandas.

```python
import os
import pandas as pd

t_test_df = pd.DataFrame([t_stat, p_val], index=['t_stat', 'p_val']).transpose()

file = os.path.join('results', 't_test_mse_results.tsv')
t_test_df.to_csv(file, sep='\t')
```